src/utils/helpers/structure/generateReviewProgress.ts

```diff
@@ -0,0 +1,86 @@
+import { ReviewResponseDecision } from '../../generated/graphql'
+
+import { FullStructure, SectionStateNEW, PageNEW } from '../../types'
+
+const generateReviewProgress = (newStructure: FullStructure) => {
+  Object.values(newStructure.sections).forEach((section) => {
+    Object.values(section.pages).forEach(generatePageReviewProgress)
+    generateSectionReviewProgress(section)
+  })
+
+  generateReviewValidity(newStructure)
+}
+
+const generateSectionReviewProgress = (section: SectionStateNEW) => {
+  section.reviewProgress = getSums(Object.values(section.pages))
+}
+
+const generatePageReviewProgress = (page: PageNEW) => {
+  const totalReviewable = page.state.filter((element) => element.isAssigned)
+  const doneConform = totalReviewable.filter(
+    (element) => element.response?.reviewResponse?.decision === ReviewResponseDecision.Approve
+  )
+  const doneNoneConform = totalReviewable.filter(
+    (element) => element.response?.reviewResponse?.decision === ReviewResponseDecision.Decline
+  )
+
+  page.reviewProgress = {
+    totalReviewable: totalReviewable.length,
+    doneConform: doneConform.length,
+    doneNoneConform: doneNoneConform.length,
+  }
+}
+
+const generateReviewValidity = (newStructure: FullStructure) => {
+  // TODO would prefer sorted pages and sorted sections available in structure
```

Comment:
Pages are already sorted since the PageNumber is the key of the map... The sections is sorted as well during first time these are fetched from the database, although since we are using the SectionCode as keys they might be out of order here. Not sure what would be the cleaver way to have these sorted. Use an array to store sections instead of an object perhaps?

Reply:
I don't think 'instead' but as a complimentary part of the structure, we could have arrays of sections and pages, in sorted manner

Reply:
Similar to what was done for `elementsById`? Is it worth doing now to use here?